learning_phase/chat_sample/client.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"strings"
+)
+
+//various commands numbered from 0, 1, 2, 3...
+const (
+	cmdauth = iota
+	cmdpm
+	cmdmsg
+	cmdquit
+)
+
+type command struct {
+	id     int
+	client *client
+	args   []string
+}
+
+type client struct {
+	conn     net.Conn
+	address  string
+	username string
+	commands chan<- command //incoming channel
```

Comment:
Did you mean outgoing channel, since messages are only written into this channel

Reply:
yeah i meant going into wrong english